web_m2x_options_manager/models/m2x_create_edit_option.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.cache import ormcache
+from odoo.tools.safe_eval import safe_eval
+
+
+class M2xCreateEditOption(models.Model):
+    _name = "m2x.create.edit.option"
+    _description = 'Manage Options "Create/Edit" For Fields'
+
+    ir_model_field_id = fields.Many2one(
+        "ir.model.fields",
+        domain=[("ttype", "in", ("many2many", "many2one"))],
+        ondelete="cascade",
+        required=True,
+        string="Field",
+    )
+
+    ir_model_field_name = fields.Char(
+        related="ir_model_field_id.name",
+        store=True,
+        string="Field Name",
+    )
+
+    ir_model_id = fields.Many2one(
+        "ir.model",
+        ondelete="cascade",
+        required=True,
+        string="Model",
+    )
+
+    ir_model_name = fields.Char(
+        compute="_compute_ir_model_name",
+        inverse="_inverse_ir_model_name",
+        store=True,
+        string="Model Name",
+    )
+
+    option_create = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create' is always set to False",
+        required=True,
+        string="Create Option",
+    )
+
+    option_create_edit = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create & Edit" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create & Edit' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create & Edit' is always set to False",
+        required=True,
+        string="Create & Edit Option",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        self.clear_caches()
```

Comment:
wouldn't be better to clear only caches of related models?

Reply:
@simahawk I need to clear the cache of this model to avoid misbehavior when calling method `_get()` which is decorated with `ormcache`

Reply:
then clear the cache of that method only ;)

Reply:
> @simahawk I need to clear the cache of this model to avoid misbehavior when calling method _get() which is decorated with ormcache

:point_up: I'd be good to put this in a comment